src/manager/Manager.sol

```diff
@@ -17,46 +18,24 @@ import { VersionedContract } from "../VersionedContract.sol";
 import { IVersionedContract } from "../lib/interfaces/IVersionedContract.sol";
 
 /// @title Manager
-/// @author Rohan Kulkarni
-/// @custom:repo github.com/ourzora/nouns-protocol 
+/// @author Neokry & Rohan Kulkarni
+/// @custom:repo github.com/ourzora/nouns-protocol
 /// @notice The DAO deployer and upgrade manager
-contract Manager is IManager, VersionedContract, UUPS, Ownable, ManagerStorageV1 {
+contract Manager is IManager, VersionedContract, UUPS, Ownable, ManagerStorageV1, ManagerStorageV2 {
     ///                                                          ///
-    ///                          IMMUTABLES                      ///
+    ///                          CONSTANTS                       ///
     ///                                                          ///
 
-    /// @notice The token implementation address
-    address public immutable tokenImpl;
+    /// @notice The count of implementation types
+    uint8 public constant IMPLEMENTATION_TYPE_COUNT = 5;
 
-    /// @notice The metadata renderer implementation address
-    address public immutable metadataImpl;
-
-    /// @notice The auction house implementation address
-    address public immutable auctionImpl;
-
-    /// @notice The treasury implementation address
-    address public immutable treasuryImpl;
-
-    /// @notice The governor implementation address
-    address public immutable governorImpl;
-
-    ///                                                          ///
-    ///                          CONSTRUCTOR                     ///
-    ///                                                          ///
-
-    constructor(
-        address _tokenImpl,
-        address _metadataImpl,
-        address _auctionImpl,
-        address _treasuryImpl,
-        address _governorImpl
-    ) payable initializer {
-        tokenImpl = _tokenImpl;
-        metadataImpl = _metadataImpl;
-        auctionImpl = _auctionImpl;
-        treasuryImpl = _treasuryImpl;
-        governorImpl = _governorImpl;
-    }
+    // Public constants for implementation types.
+    // Allows for adding new types later easily compared to a enum.
```

Comment:
i believe at this point enums under the hood are uint8 – this just makes it more clear when you add items.